src/P/List.hs

```diff
@@ -45,3 +46,11 @@ lastMaybe = listToMaybe . reverse
 -- | count the number of elements satisfying a predicate in a list
 count :: (a -> Bool) -> [a] -> Int
 count predicate = length . filter predicate
+
+safeIndex :: [a] -> Int -> Maybe a
```

Comment:
Actually, why not provide both `safeIndex` and `(!?)` and implement the second as:
```
(!?) :: [a] -> Int -> Maybe a
(!?) = safeIndex
```
People are more likely to use the `(!?)` operator, but when they go looking for something they are more likely to hoogle for `safeIndex`. I'm pretty sure thats how @novemberkilo found that function in Deckard.

I'm pretty